hw4/04-templates.js

```diff
@@ -3,62 +3,118 @@ const axios = require('axios');
 
 const app = express();
 const port = process.env.PORT || 5001;
+const url = 'https://restcountries.com/v3.1/all';
+
 
-// Use Pug as the templating engine
 app.set('views', __dirname + '/views');
 app.set('view engine', 'pug');
 
-// REST Countries URL
-const url = 'https://restcountries.com/v3.1/all';
 
-// Add your code here
+let groupBy = function(data, key) {
+  return data.reduce(function(storage, item) {
+    let group = item[key];
+
+    storage[group] = storage[group] || [];
+    storage[group] = Number(storage[group]) + item.population;
+
+    return storage; 
+  }, {});
+};
+
+
+const instructions = [
+  { 'title'     : '/capitals',
+    'params'    : 'capital',
+    'command'   : '`${country.name.official} - ${country.capital}`',
```

Comment:
There are countries with no capital (Eg. Antarctica). The message needs to say `no data` like mentioned on the HW document. Also the capital key has an array of strings but you seem to just use `country.capital`. Did this code run and display the capital properly?